File: SegundaParte/Semana11/C++/Notas.cpp
#include <iostream>
#include <stdexcept> // para std::runtime_error

int main() {
    const int NUM_ALUNOS = 10;
    double vet[NUM_ALUNOS];
    double acum = 0;

    std::cout << "Calcular a media de 10 notas." << std::endl;
    std::cout << "Para simplificar, todo mundo tirou 10!!!" << std::endl;

    try {
        for (int cont = 0; cont <= NUM_ALUNOS; ++cont)
            vet[cont] = 10;
    } catch (const std::exception& e) {
        std::cerr << "Ups! Entrei uma nota que nao devia!" << std::endl;
        std::cerr << "Note: nao ha tratamento" << std::endl;
    }

    std::cout << "Observe a mesma falha, agora sem captura." << std::endl;
    
    for (int cont = 0; cont <= NUM_ALUNOS; cont++)
        acum += vet[cont];

    std::cout << "Media = " << acum / NUM_ALUNOS << std::endl;

    return 0;
}

File: SegundaParte/Semana11/Notas.java
public class Notas {
  public static void main(String[] args) {
   final int NUM_ALUNOS = 10;
   double [] vet = new double [NUM_ALUNOS] ;
   double acum = 0;
   System.out.println("Calcular a media de 10 notas.");
   System.out.println("Para simplificar, todo mundo tirou 10!!!");
   try {
     for( int cont = 0 ; cont <= NUM_ALUNOS ; ++cont )
     vet[cont] = 10;
   } catch( Exception e ) {
     System.out.print("Ups! Entrei uma nota que nao devia!");
     System.out.println(" Note: nao ha tratamento");
   }
   System.out.println("Observe a mesma falha, agora sem captura.");
   for( int cont = 0 ; cont <= NUM_ALUNOS ; cont++ )
     acum += vet[cont];
   System.out.println("Media = " + acum/NUM_ALUNOS);
 }
}

